docs/src/main/asciidoc/security-keycloak-admin-client.adoc

```diff
@@ -85,16 +87,61 @@ public class RolesResource {
 }
 ----
 
-Now, you can produce the `Keycloak` bean yourself, or you can instruct Quarkus to produce `Keycloak` bean.
+<1> Create  a default Keycloak Admin Client which can perform Keycloak `master` realm administration tasks as an `admin-cli` client, such as adding new realms, clients and users.
+
+The only configuration which is required to create this Keycloak Admin Client is a Keycloak server URL, for example:
+
+[source,properties]
+----
+quarkus.keycloak.admin-client.server-url=http://localhost:8081
+----
+
+Injecting Keycloak Admin Client instead of creating it directly in the application code is a simpler and more flexible option but you can create the same admin client manually if necessary:
+
+[source,java]
+----
+package org.acme.keycloak.admin.client;
+
+import org.keycloak.admin.client.Keycloak;
+import org.keycloak.admin.client.KeycloakBuilder;
+import org.keycloak.representations.idm.RoleRepresentation;
+
+import javax.annotations.PostConstruct;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import java.util.List;
+
+@Path("/api/admin")
+public class RolesResource {
+
+        Keycloak keycloak;
+
+        @PostConstruct
+        public void initKeycloak() {
```

Comment:
Please don't forget to close the bean in `@PreDestroy`. Also resources are by default `@javax.inject.Singleton` in classic RESTEasy (unless you set `quarkus.resteasy.singleton-resources`) and if I read `io.quarkus.resteasy.reactive.server.deployment.ResteasyReactiveCDIProcessor#beanDefiningAnnotations` correctly, in RR it's always Singleton (maybe you can annotate resource with `@RequestScoped`). As `Keycloak` is `AutoCloseable`, I would close it when I don't need it.

Reply:
@michalvavrik Makes sense, I'll just add `PreDestroy`